### PR TITLE
Initial manifest for io.github.janbar.noson

### PR DESCRIPTION
**Pull request by cosimoc**
QT Sonos client for Linux.
It's missing an appdata file and screenshots, but  @nedrichards volunteered to possibly help with those :-)

### CODE REVIEW

File: io.github.janbar.noson.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  "build-options": {
+    "cflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "cxxflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "ldflags": "-fstack-protector-strong -Wl,-z,relro,-z,now -L/app/lib",
+    "env": {
+      "V": "1"
+    }
+  },
+  "modules": [
+    "qt5.json",
```
Comment (TingPing) [inline io.github.janbar.noson.json]: Why not just use the kde runtime?

Comment (cosimoc) [reply io.github.janbar.noson.json]: I guess I could... I was copying what VLC did, but maybe it would be less churn to use the KDE runtime directly.

Comment (barthalion) [inline io.github.janbar.noson.json]: Use KDE runtime instead.

File: io.github.janbar.noson.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    "/share/man",
+    "/share/pkgconfig",
+    "/lib/cmake",
+    "*.la",
+    "*.a"
+  ],
+  "build-options": {
+    "cflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "cxxflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "ldflags": "-fstack-protector-strong -Wl,-z,relro,-z,now -L/app/lib",
```
Comment (barthalion) [inline io.github.janbar.noson.json]: There is ld.so.conf in place that adds /app/lib already, is it not working properly?

File: io.github.janbar.noson.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      "name": "noson",
+      "buildsystem": "cmake",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/janbar/noson-app.git",
+          "tag": "3.3.4"
+        },
+        {
+          "type": "patch",
```
Comment (barthalion) [inline io.github.janbar.noson.json]: Would be nice to have these sent upstream.

File: io.github.janbar.noson.json [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        }
+      ]
+    },    
+    {
+      "name": "noson",
+      "buildsystem": "cmake",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/janbar/noson-app.git",
```
Comment (barthalion) [inline io.github.janbar.noson.json]: Has upstream author been contacted about intent to add it to Flathub?

File: io.github.janbar.noson.json (hunk 3 above)
Comment (barthalion) [reply io.github.janbar.noson.json]: Ah, just noticed that they were. Never mind. :+1:

**Comment by barthalion** [comment]
Hi @janbar, would you be interested in maintaining Flathub package "officially"? We would be happy to give you push access so you can update it for new releases.

**Comment by janbar** [comment]
Yes sure. But currently I haven't skill with flatpak.

**Comment by cosimoc** [comment]
Fixed your comments, @barthalion and @TingPing.

File: io.github.janbar.noson.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    "cxxflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "ldflags": "-fstack-protector-strong -Wl,-z,relro,-z,now",
+    "env": {
+      "V": "1"
+    }
+  },
+  "modules": [
+    {
+      "name": "noson",
+      "buildsystem": "cmake",
```
Comment (eszlari) [inline io.github.janbar.noson.json]:
```
"buildsystem": "cmake-ninja",
"builddir": true,
"config-opts": [
  "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
]
```

File: io.github.janbar.noson.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{
+  "app-id": "io.github.janbar.noson",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.10",
```
Comment (eszlari) [inline io.github.janbar.noson.json]: Did you try 5.11?

Comment (nedrichards) [reply io.github.janbar.noson.json]: 5.11 hasn't been shipped yet, has it?

**Comment by nedrichards** [comment]
I'll happily add some appdata and look at sending that upstream for a future release as well.

File: io.github.janbar.noson.json (hunk 6 above)
Comment (Alexander-Wilms) [reply io.github.janbar.noson.json]: It's available to install from Flathub

Comment (nedrichards) [reply io.github.janbar.noson.json]: Doesn't build against 5.11:
> CMake Error at gui/CMakeLists.txt:42 (qt5_use_modules):
  Unknown CMake command "qt5_use_modules".

**Comment by nedrichards** [comment]
@janbar before we push this does the app-id make sense to you? There's more information about this here if you need: http://docs.flatpak.org/en/latest/conventions.html#application-ids

File: io.github.janbar.noson.json (hunk 6 above)
Comment (eszlari) [reply io.github.janbar.noson.json]: > Doesn't build against 5.11:

Fixing this seems to simple: https://phabricator.kde.org/D12272

**Comment by janbar** [comment]
@nedrichards the app id seems good for me. I updated the var APP_ID with this new value.

**Comment by nedrichards** [comment]
@janbar great! looking forward to dropping some of these changes when the new version comes out.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/io.github.janbar.noson

**Pull request closed without merging**